File: Banda_Beat_Android/src/com/irontec/bandabeat/db/Track.java
package com.irontec.bandabeat.db;

public class Track {
	
	private long idTrack;
	private String album;
	private String grupo;
	private String imageMini;
	private String imageThumb;
	private String imageBig;
	private String imageProfile;
	private String titulo;
	private String url;
	private String token;
	private String duration;
	
	
	public long getIdTrack() {
		return idTrack;
	}
	public void setIdTrack(long idTrack) {
		this.idTrack = idTrack;
	}
	public String getAlbum() {
		return album;
	}
	public void setAlbum(String album) {
		this.album = album;
	}
	public String getGrupo() {
		return grupo;
	}
	public void setGrupo(String grupo) {
		this.grupo = grupo;
	}
	public String getImageMini() {
		return imageMini;
	}
	public void setImageMini(String imageMini) {
		this.imageMini = imageMini;
	}
	public String getImageThumb() {
		return imageThumb;
	}
	public void setImageThumb(String imageThumb) {
		this.imageThumb = imageThumb;
	}
	public String getImageBig() {
		return imageBig;
	}
	public void setImageBig(String imageBig) {
		this.imageBig = imageBig;
	}
	public String getImageProfile() {
		return imageProfile;
	}
	public void setImageProfile(String imageProfile) {
		this.imageProfile = imageProfile;
	}
	public String getTitulo() {
		return titulo;
	}
	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}
	public String getUrl() {
		return url;
	}
	public void setUrl(String url) {
		this.url = url;
	}
	public String getToken() {
		return token;
	}
	public void setToken(String token) {
		this.token = token;
	}
	public String getDuration() {
		return duration;
	}
	public void setDuration(String duration) {
		this.duration = duration;
	}	
}

File: Banda_Beat_iOS/Banda Beat/Track.m
//
//  Track.m
//  Banden Lehia
//
//  Created by Iker Mendilibar on 30/10/12.
//  Copyright (c) 2012 Irontec S.L. All rights reserved.
//

#import "Track.h"
#import "NSString+IMAddition.h"

@implementation Track

@dynamic album;
@dynamic grupo;
@dynamic idTrack;
@dynamic imageBig;
@dynamic imageThumb;
@dynamic imageProfile;
@dynamic imageMini;
@dynamic imageIphone;
@dynamic titulo;
@dynamic url;
@dynamic token;
@dynamic favorite;
@dynamic playlist;
@dynamic duration;
@synthesize order;

-(void)setDataWithdictionary:(NSDictionary*)data order:(NSInteger)ord
{
    NSString *name= [data objectForKey:@"titulo"];
    
    if ([NSString isEmptyString:name]) {
        name = @"";
    }
    
    [self setTitulo:name];
    [self setAlbum:[data objectForKey:@"album"]];
    [self setGrupo:[data objectForKey:@"grupo"]];
    [self setIdTrack:[data objectForKey:@"idTrack"]];
    [self setImageBig:[data objectForKey:@"imageBig"]];
    [self setImageProfile:[data objectForKey:@"imageProfile"]];
    [self setImageMini:[data objectForKey:@"imageMini"]];
    [self setImageThumb:[data objectForKey:@"imageThumb"]];
    [self setImageIphone:[data objectForKey:@"imageiPhone"]];
    [self setUrl:[data objectForKey:@"url"]];
    [self setToken:[data objectForKey:@"token"]];
    [self setDuration:[data objectForKey:@"duration"]];
    [self setOrder:[NSNumber numberWithInteger:ord]];
    [self setFavorite:[NSNumber numberWithBool:NO]];
}


@end
